src/main/webapp/utilities/EventManager.js

```diff
@@ -0,0 +1,118 @@
+export class EventManager {
+    /**
+     * Constructor for normal usage
+     * @constructor
+     */
+    constructor() {
+        this.localStorage = Window.localStorage;
+
+        setup();
+    }
+
+    /**
+     * Constructor that enables injecting a mock object (for testing)
+     * @constructor
+     * @param {*} LocalStorageObject the mock object
+     */
+    constructor(LocalStorageObject) {
+        this.localStorage = localStorageObject;
+
+        setup();
+    }
+
+    /**
+     * Gets JSON-parsed value from LocalStorage
+     * @returns {Any} the value on that field
+     */
+    getValueInLocalStorageParsed() {
+        return JSON.parse(this.localStorage.getItem('events'));
+    }
+
+    /**
+     * Sets value to LocalStorage
+     * @param {Any} value the value to be set (if not string -> convert to one using JSON)
+     * @returns {void}
+     */
+    setValueInLocalStorage(value) {
+        if(typeof value === 'string') {
+            this.localStorage.setItem('events', value);
+        } else {
+            const jsonSerialized = JSON.stringify(value);
+
+            this.localStorage.setItem('events', jsonSerialized);
+        }
+    }
+
+    /**
+     * Initial setup of the LocalStorage
+     * @returns {void}
+     */
+    setup() {
+        this.eventsList = this.getValueInLocalStorageParsed();
```

Comment:
I believe you shouldn't store it in a instance field but only in local storage.

Reply:
I think it's worth it, because it makes stuff easier to understand and the additional redundancy is too small to hinder performance or anything else!

Reply:
Opening two instances of application in separate browser tabs will lead to surprising behaviour.

Reply:
As this is an MVP can we keep it like this? I think to solve this, we could add something like pooling and frequently query the LocalStorage. I think that issues has nothing to do with our approach here, rather than the fact that we don't have any mechanism of live communication (like websockets) in place. Idk, maybe I am wrong. What solution do you propose?

Reply:
Or you can remove unnecessary code that caches value from local storage that will solve the problem without complications.

Reply:
Fixed! Can you please have a look?